### PR TITLE
Fix URL Concat Issues

### PR DESCRIPTION
**Pull request by EpicKris**
Use `getUrl` to build the URL correctly (fixes issue with base URL possibly containing or not containing a slash at the end of config)

### CODE REVIEW

File: src/energenieApi/energenieApi.ts [hunk 1]
```diff
@@ -53,14 +54,14 @@ export class EnergenieApi {
   public toggleSocketPower(id: number, action: string): Promise<void> {
     const params = `params={ "id": ${id} }`;
 
-    return this.makeHttpRequest(() => this.httpClient.post(`${this.baseUrl}subdevices/` + action, params),
+    return this.makeHttpRequest(() => this.httpClient.post(`${getUrl('subdevices', this.baseUrl)}` + action, params),
```
Comment (jamesrodda) [inline src/energenieApi/energenieApi.ts:57]: Nice, getUrl is very handy. Could you pull the calls out into variables i.e. `const url = getUrl('subdevices', this.baseUrl) + action;`. Just makes it a bit easier to read.

Comment (jamesrodda) [reply src/energenieApi/energenieApi.ts:57]: You also don't need the string interpolation `${}` when you don't have a string literal.

Comment (jamesrodda) [reply src/energenieApi/energenieApi.ts:57]: Whilst we're looking at this too, the "action" parameter should probably be removed from `platformAccessory` - the specific values required are an api return, we should decouple here.

File: src/energenieApi/energenieApi.ts [hunk 2]
```diff
@@ -53,14 +54,14 @@ export class EnergenieApi {
...
       `toggleSocketPower/${action}`);
   }
 
   public getSubdeviceInfo(id: number): Promise<SubDevice> {
     const params = `params={ "id": ${id} }`;
 
-    return this.makeHttpRequest(() => this.httpClient.post(`${this.baseUrl}subdevices/show`, params),
+    return this.makeHttpRequest(() => this.httpClient.post(`${getUrl('subdevices/show', this.baseUrl)}`, params),
```
Comment (jamesrodda) [inline src/energenieApi/energenieApi.ts:64]: Same points as above

File: src/energenieApi/energenieApi.ts (hunk 1 above)
Comment (EpicKris) [reply src/energenieApi/energenieApi.ts]: Great suggestions! Implemented.

**Comment by EpicKris** [comment]
Thanks again for the review @jamesrodda, implemented.

File: src/energenieApi/energenieApi.ts [hunk 3]
```diff
@@ -50,17 +51,20 @@ export class EnergenieApi {
       'getSubDevices');
   }
 
-  public toggleSocketPower(id: number, action: string): Promise<void> {
+  public toggleSocketPower(id: number, value: boolean): Promise<void> {
     const params = `params={ "id": ${id} }`;
+    const action = value ? 'power_on' : 'power_off';
+    const url = getUrl('subdevices' + action, this.baseUrl);
```
Comment (jamesrodda) [inline src/energenieApi/energenieApi.ts:57]: Think this is missing a slash, should be `` `subdevices/${action}` ``

Comment (EpicKris) [reply src/energenieApi/energenieApi.ts]: Fixed.

**Comment by jamesrodda** [comment]
> Thanks again for the review @jamesrodda, implemented.

No probs - some great additions here, much appreciated 👍

**Pull request merged**